In [71]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

In [72]:
# Read in the raw_counts.txt file
raw_counts = pd.read_csv("../../02_data/processed-data/raw_counts.txt", delimiter="\t", comment="#")

In [73]:
# Check the dimensions of the data frame
raw_counts.shape

(78899, 33)

In [74]:
raw_counts.head(5)

Geneid                                                Chr  \
0  ENSG00000142611  1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;...   
1  ENSG00000284616                                            1;1;1;1   
2  ENSG00000260972                                                  1   
3  ENSG00000229280                                                  1   
4  ENSG00000232596  1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;...   

                                               Start  \
0  3069168;3069183;3069197;3069203;3069211;318612...   
1                    5301928;5303328;5304401;5306942   
2                                            5492978   
3                                            4175528   
4  4571481;4571489;4571499;4571504;4571504;457151...   

                                                 End  \
0  3069296;3069296;3069296;3069296;3069296;318647...   
1                    5302004;5303393;5305029;5307394   
2                                            5494674   
3                                            4175899   
4  4571629;4571629;4571629;4571629;4571629;457162...   

                                              Strand  Length  \
0  +;+;+;+;+;+;+;+;+;+;+;+;+;+;+;+;+;+;+;+;+;+;+;...    9767   
1                                            -;-;-;-    1225   
2                                                  +    1697   
3                                                  -     372   
4  +;+;+;+;+;+;+;+;+;+;+;+;+;+;+;+;+;+;+;+;+;+;+;...    3328   

   ../../02_data/genome-alignment/SRR10983637.trim.sorted.bam  \
0                                                  0            
1                                                  0            
2                                                  0            
3                                                  0            
4                                                  0            

   ../../02_data/genome-alignment/SRR10983638.trim.sorted.bam  \
0                                                  1            
1                                                  0            
2                                                  0            
3                                                  0            
4                                                  0            

   ../../02_data/genome-alignment/SRR10983639.trim.sorted.bam  \
0                                                  0            
1                                                  0            
2                                                  0            
3                                                  0            
4                                                  0            

   ../../02_data/genome-alignment/SRR10983640.trim.sorted.bam  ...  \
0                                                  0           ...   
1                                                  0           ...   
2                                                  0           ...   
3                                                  0           ...   
4                                                  0           ...   

   ../../02_data/genome-alignment/SRR10983654.trim.sorted.bam  \
0                                                  7            
1                                                  0            
2                                                  0            
3                                                  0            
4                                                  0            

   ../../02_data/genome-alignment/SRR10983655.trim.sorted.bam  \
0                                                  0            
1                                                  0            
2                                                  0            
3                                                  0            
4                                                  0            

   ../../02_data/genome-alignment/SRR10983656.trim.sorted.bam  \
0                                                 36            
1       

The dimension of the data frame is 78899 rows and 33 columns, our total number of samples were 27, hence we can remove the 6 remaining columns.

In [75]:
# Set the rowname of the data frame to gene ids
raw_counts = raw_counts.set_index("Geneid")

# Create a new data frame and only keep the columns corresponding to the different samples in our dataset
counts = raw_counts.iloc[:,5:] # df.iloc[rows, columns]

# Check the new data frame
counts.head(5)

../../02_data/genome-alignment/SRR10983637.trim.sorted.bam  \
Geneid                                                                        
ENSG00000142611                                                  0            
ENSG00000284616                                                  0            
ENSG00000260972                                                  0            
ENSG00000229280                                                  0            
ENSG00000232596                                                  0            

                 ../../02_data/genome-alignment/SRR10983638.trim.sorted.bam  \
Geneid                                                                        
ENSG00000142611                                                  1            
ENSG00000284616                                                  0            
ENSG00000260972                                                  0            
ENSG00000229280                                                  0            
ENSG00000232596                                                  0            

                 ../../02_data/genome-alignment/SRR10983639.trim.sorted.bam  \
Geneid                                                                        
ENSG00000142611                                                  0            
ENSG00000284616                                                  0            
ENSG00000260972                                                  0            
ENSG00000229280                                                  0            
ENSG00000232596                                                  0            

                 ../../02_data/genome-alignment/SRR10983640.trim.sorted.bam  \
Geneid                                                                        
ENSG00000142611                                                  0            
ENSG00000284616                                                  0            
ENSG00000260972                                                  0            
ENSG00000229280                                                  0            
ENSG00000232596                                                  0            

                 ../../02_data/genome-alignment/SRR10983641.trim.sorted.bam  \
Geneid                                                                        
ENSG00000142611                                                  0            
ENSG00000284616                                                  0            
ENSG00000260972                                                  0            
ENSG00000229280                                                  0            
ENSG00000232596                                                  0            

                 ../../02_data/genome-alignment/SRR10983642.trim.sorted.bam  \
Geneid                                                                        
ENSG00000142611                                                  0            
ENSG00000284616                                                  0            
ENSG00000260972                                                  0            
ENSG00000229280                                                  0            
ENSG00000232596                                                  0            

                 ../../02_data/genome-alignment/SRR10983643.trim.sorted.bam  \
Geneid                                                                        
ENSG00000142611                                                  0            
ENSG00000284616                                                  0            
ENSG00000260972                                                  0            
ENSG00000229280                                                  0            
ENSG00000232596                                                  0            

                 ../../02_data/genome-alignment/SRR10983644.trim.sorted.bam  \
Geneid                                                      

After we have gotten rid of the unecessary columns, now we focus on renaming the columns so that each column is named as the run accession for the sample.

In [76]:
# Use python regular expression to rename the column to their suitable name.
counts.columns = counts.columns.str.extract(r"(SRR\d+)", expand=False)

# Remove the index name
counts.index.name = None

# Check the new column names
counts.columns

Index(['SRR10983637', 'SRR10983638', 'SRR10983639', 'SRR10983640',
       'SRR10983641', 'SRR10983642', 'SRR10983643', 'SRR10983644',
       'SRR10983645', 'SRR10983646', 'SRR10983647', 'SRR10983648',
       'SRR10983649', 'SRR10983650', 'SRR10983651', 'SRR10983652',
       'SRR10983653', 'SRR10983654', 'SRR10983655', 'SRR10983656',
       'SRR10983657', 'SRR10983658', 'SRR10983659', 'SRR10983660',
       'SRR10983661', 'SRR10983662', 'SRR10983663'],
      dtype='object')

In [ ]:
# Double check the number of samples
len(counts.columns)

27

In [ ]:
# Write out the counts data frame as a csv file

# Source - https://stackoverflow.com/a
# Posted by Andy Hayden, modified by community. See post 'Timeline' for change history
# Retrieved 2025-12-15, License - CC BY-SA 4.0
counts.to_csv("../../02_data/processed-data/counts.csv", sep='\t', encoding='utf-8', index=True, header=True)
